src/components/photo_input_wrapper.tsx

```diff
@@ -39,44 +42,34 @@ const PhotoInputWrapper: FC<PhotoInputWrapperProps> = ({
     const [loading, setLoading] = useState(false) // Loading state
     const [error, setError] = useState('') // Loading state
 
-    const img = new Image()
-    const canvas = document.createElement('canvas')
-    const ctx = canvas.getContext('2d') as CanvasRenderingContext2D
-
-    function compressJpegBlob(jpegBlob: Blob, quality = 0.7) {
-        return new Promise((resolve, reject) => {
-            // Load the JPEG blob into the image
-            img.onload = () => {
-                // Set canvas dimensions based on the loaded image
-                canvas.width = Math.min(img.width, MAX_IMAGE_DIM_WIDTH)
-                canvas.height = Math.min(img.height, MAX_IMAGE_DIM_HEIGHT)
+    /**
+     * Compresses an image file (Blob) while maintaining its aspect ratio and ensuring it does not exceed specified size limits.
+     *
+     * @param {Blob} imageBlob - The original image file as a Blob object that needs to be compressed.
+     *
+     * @returns {Promise<Blob | undefined>} - A Promise that resolves to the compressed image file as a Blob.
+     *                                         If an error occurs during compression, it will be caught,
+     *                                         and the function may return undefined.
+     *
+     * @throws {Error} - Throws an error if the compression process fails.
+     *
+     */
+    async function compressFile(imageBlob: Blob) {
+        /*The compressed file will have a maximum size of `maxSizeMB` 
+         and dimensions constrained by`maxWidthOrHeight`. */
+        const options = {
+            maxSizeMB: 0.2,
+            useWebWorker: true,
+            maxWidthOrHeight: 1950,
```

Comment:
Should this be `Math.max(MAX_IMAGE_DIM_WIDTH, MAX_IMAGE_DIM_HEIGHT)`?

Reply:
Thanks for catching this. I have updated it accordingly.